resolve-aws-secret-version/src/kubernetes.ts

```diff
@@ -0,0 +1,53 @@
+export type KubernetesObject = {
+  kind: string
+}
+
+export const isKubernetesObject = (a: unknown): a is KubernetesObject =>
+  typeof a === 'object' && a !== null && 'kind' in a && typeof a.kind === 'string'
+
+export type KubernetesAWSSecret = KubernetesObject & {
+  metadata: {
+    name: string
+  }
+  spec: {
+    stringDataFrom: {
+      secretsManagerSecretRef: {
+        secretId: string
+        versionId: string
+      }
+    }
+  }
+}
+
+export function assertKubernetesAWSSecret(a: KubernetesObject): asserts a is KubernetesAWSSecret {
```

Comment:
Basically same, but it throws an error on validation failure.
https://github.com/quipper/monorepo-deploy-actions/blob/73e52f3e6a35854e436d9bf83393e393db99b5fc/resolve-aws-secret-version/src/resolve.ts#L93-L110